### ISSUE TITLE
Gettext customization option to not wrap output doesn't work

### ISSUE THREAD

**Issue by hhyyrylainen** [WeblateOrg/weblate#6688]
**Describe the issue**

Weblate really likes to disagree with the command line gettext tools regarding how to wrap text. I've finally had enough and I changed the option in the Weblate gettext output customization plugin to be "just wrap on newline" as that seems to be according to the documentation https://docs.weblate.org/en/weblate-4.8.1/admin/addons.html#addon-weblate-gettext-customize what the gettext command line tools do when wrapping is disabled in them. However, that seems to have made things worse as I just got a PR from Weblate that seems to have the default wrapping option enabled: https://github.com/Revolutionary-Games/Thrive/pull/2662/files (also I just noticed that for some reason Weblate wants to resurrect outdated comments in that PR, I'll open a separate issue about that, which I've seen before but as I recently reset the repository on Weblate, I hoped it would be gone...)

**I already tried**

Describe the steps you tried to solve the problem yourself.

- [x] I've read and searched [the docs](https://docs.weblate.org/) and did not find the answer there.
      If you didn’t try already, try to search there what you wrote above. I linked the docs above
- [x] found this issue where basically the column 77 wrapping is different between weblate and the gettext tools (this is the underlying problem I have): https://github.com/WeblateOrg/weblate/issues/6350 and it was said it is not fixable, so I decided to use this no-wrapping mode

**To Reproduce the issue**
This is what I did after trying to use the column 77 wrap option (and updating Weblate a month or so ago) for many months, and consistently getting PRs with changes to how text is wrapped as we very regularly run the gettext tools to add new text strings.
 
So this is how I tried to move to the new mode:
- committed and sent latest updates from weblate
- locked editing on weblate
- merged the PR from weblate
- changed the plugin option on weblate
- and unlocked editing after weblate updated from the main repository
- and now after a few days weblate has made a PR of the pending changes and I see what mess the PR is.

I did not try to set the plugin to no wrap at all mode, as that wouldn't also match what gettext does, right?

**Expected behavior**

I expect that weblate when configured to not wrap things, *wouldn't* wrap things.

**Screenshots**

![Kuvakaappaus - 2021-10-17 21-58-15](https://user-images.githubusercontent.com/3796411/137641248-a478ccdc-bee5-4f54-9ade-38c2ac8a9b51.png)

**Server configuration and status**

Weblate installation: PyPI

```
 * Weblate: 4.8.1
 * Django: 3.2.6
 * siphashc: 2.1
 * translate-toolkit: 3.3.6
 * lxml: 4.6.3
 * Pillow: 8.1.2
 * bleach: 3.3.0
 * python-dateutil: 2.8.1
 * social-auth-core: 4.1.0
 * social-auth-app-django: 5.0.0
 * django-crispy-forms: 1.11.2
 * oauthlib: 3.1.0
 * django-compressor: 2.4
 * djangorestframework: 3.12.4
 * django-filter: 2.4.0
 * django-appconf: 1.0.4
 * user-agents: 2.2.0
 * filelock: 3.0.12
 * setuptools: 49.1.3
 * jellyfish: 0.8.2
 * openpyxl: 3.0.7
 * celery: 5.0.5
 * kombu: 5.0.2
 * translation-finder: 2.9
 * weblate-language-data: 2021.5
 * html2text: 2020.1.16
 * pycairo: 1.20.0
 * pygobject: 3.40.1
 * diff-match-patch: 20200713
 * requests: 2.25.1
 * django-redis: 4.12.1
 * hiredis: 2.0.0
 * sentry_sdk: 1.0.0
 * Cython: 0.29.22
 * misaka: 2.1.1
 * GitPython: 3.1.14
 * borgbackup: 1.1.16
 * pyparsing: 2.4.7
 * pyahocorasick: 1.4.2
 * python-redis-lock: 3.7.0
 * Python: 3.9.2
 * Git: 2.31.1
 * psycopg2-binary: 2.8.6
 * chardet: 4.0.0
 * ruamel.yaml: 0.17.2
 * boto3: 1.17.44
 * zeep: 4.0.0
 * aeidon: 1.9
 * Redis server: 6.2.6
 * PostgreSQL server: 13.4
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: mail.revolutionarygamesstudio.com
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://localhost:6379, redis://localhost:6379, regular
 * Platform: Linux 5.14.9-200.fc34.x86_64 (x86_64)

```

**Weblate deploy checks**

```
System check identified some issues:

WARNINGS:
?: (weblate.W025.fluent) Failure in loading handler for fluent file format: No module named 'translate.storage.fluent'
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#optional-deps
?: (weblate.W025.ini) Failure in loading handler for ini file format: Missing iniparse library.
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#optional-deps
?: (weblate.W025.islu) Failure in loading handler for islu file format: Missing iniparse library.
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#optional-deps
?: (weblate.W025.laravel) Failure in loading handler for laravel file format: No module named 'phply'
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#optional-deps
?: (weblate.W025.php) Failure in loading handler for php file format: No module named 'phply'
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#optional-deps
?: (weblate.W025.stringsdict) Failure in loading handler for stringsdict file format: No module named 'translate.storage.stringsdict'
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#optional-deps
?: (weblate.W033.Gerrit) Failure in loading VCS module for Gerrit: git: 'review' is not a git command. See 'git --help'.
 (1)
        HINT: https://docs.weblate.org/en/weblate-4.8.1/vcs.html
?: (weblate.W033.Mercurial) Failure in loading VCS module for Mercurial: [Errno 2] No such file or directory: 'hg'
        HINT: https://docs.weblate.org/en/weblate-4.8.1/vcs.html
?: (weblate.W033.Subversion) Failure in loading VCS module for Subversion: git: 'svn' is not a git command. See 'git --help'.

The most similar commands are
        fsck
        mv
        show
 (1)
        HINT: https://docs.weblate.org/en/weblate-4.8.1/vcs.html

INFOS:
?: (weblate.I021) Error collection is not set up, it is highly recommended for production use
        HINT: https://docs.weblate.org/en/weblate-4.8.1/admin/install.html#collecting-errors

System check identified 10 issues (1 silenced).

```

**Additional context**

**Comment by nijel** [comment]
The add-on configuration in https://github.com/WeblateOrg/weblate/issues/6688#issuecomment-952983845 shows configuration to "Wrap lines at 77 characters and at newlines". Are you sure it was set to "Only wrap lines at newlines" when the changes were committed in Weblate? The setting only takes effect at commit time - it does not change files which were already committed before.

**Comment by hhyyrylainen** [comment]
I changed the option before I did a repo reset in weblate repo maintenance. Also I think I had the PR open for at least a couple of days, which now that I'm looking at the last language with changes in there *seems* like it may have worked correctly: https://github.com/Revolutionary-Games/Thrive/pull/2662/commits/6be24a4c992078bd810035c5eaecd5fb16e60473
However, I'm left wondering why would weblate mess up the initial languages, because as I mentioned, I went about doing this so that I first pushed pending changes from weblate, merged those, changed the option in weblate, and reset the weblate repo. And then I think I merged in my changes for changing the options for gettext that were used (these last two steps may have been the other way around). So I'm left wondering why first commits in that PR ended up with wrong settings. I guess this does give me hope that despite the initial problems if we stuck with this approach, weblate would eventually get it and it would stop being a problem.
